superset/assets/src/visualizations/core/components/SuperChart.jsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import getChartComponentRegistry from '../registries/ChartComponentRegistrySingleton';
+import getChartTransformPropsRegistry from '../registries/ChartTransformPropsRegistrySingleton';
+
+const STATUS = {
+  IDLE: 1,
+  LOADING: 2,
+  SUCCESS: 3,
+  FAILURE: 4,
+};
+const IDENTITY = x => x;
+
+const propTypes = {
+  id: PropTypes.string,
+  className: PropTypes.string,
+  chartProps: PropTypes.object.isRequired,
```

Comment:
I know the ChartProps now define this API well, any use in defining a shape here?

Reply:
oh right. I will check if it is instanceof `ChartProps` then.